### DISCUSSION TITLE
How to Extend Directus System Collections

### DISCUSSION THREAD

**Discussion by philleepflorence** [directus/directus#6356] [Archive]
> Currently mapping out data store extensions to D9 and looking for ideas on an efficient methods for extending or scaling directus system collections.

Welcome any ideas, options, use cases, or unseen caveats.
Thanks!

### Use Cases
- Add `addl. contact information` or `preferences` to Directus Users.
- Add `instructions`, `notes`, or `tasks` to Directus Collections.

### Related Discussions
- Mapping of system tables and columns to custom ones - #4900

**Comment by philleepflorence** [comment]
### 1. Schema Update Option

- Useful for few fields.
    - What is the chance of future field changes or clashes?

**Reply by philleepflorence** [reply]
Keeping track of related discussions:
#6404 - Allow editor to add new fields to files collection

**Comment by philleepflorence** [comment]
### Related Collection Option
- Preserve Directus [Data Integrity](https://en.wikipedia.org/wiki/Data_integrity).
- Leveage [Relational Model](https://en.wikipedia.org/wiki/Relational_model).
    - Manage via [Fields: Create a Field](https://docs.directus.io/reference/api/system/relations/#create-a-relation)
    - Manage via [Relations: Create a Relation](https://docs.directus.io/reference/api/system/relations/#create-a-relation)


#### Relationship via Directus Relations
| DIRECTUS COLLECTIONS      | RELATED FIELDS                          | EXTENDED COLLECTIONS        |
| :--                                          | :--                                                    | :--                           |
| Users                                 | User PK UUID                        | Users Meta            |
| --  UUID                                 | Meta FK UUID                       | -- user                   |
| Collections                        | Collection PK VARCHAR      | Collections Meta      |
| --  collection                               | Meta FK VARCHAR                                  | --   collection                 |

**Comment by rijkvanzanten** [comment]
Just go into Settings > Data Model, enable "System Collections" in the right sidebar, and add fields to your heart's content ✨ 

The chance of Directus implementing a new meta field that happens to have the exact same name as a custom field exists, but is next to none

**Reply by philleepflorence** [reply]
Thanks!
So you suggest [Schema Update Option](https://github.com/directus/directus/discussions/6356#discussioncomment-884770)?
Should work for singleton fields.

---

How about this use case?
> Add `instructions`, `notes`, or `tasks` to Directus Collections - `/settings/data-model/directus_collections`.

Will that change the settings UI for other collections?
e.g. `/settings/data-model/pages` - Interface for `instructions` under *Collection Setup*.
Not seeing the layout in Directus Fields...
